assets/src/edit-story/utils/useRefreshPostEditURL.js

```diff
@@ -18,26 +18,22 @@
  * External dependencies
  */
 import { useCallback } from 'react';
-import { addQueryArgs } from '@web-stories-wp/design-system';
 
 /**
  * Update page URL in browser.
  *
  * @param {number} postId Current story id.
+ * @param {string} postEditURL Current story's edit link.
  * @return {Function} Function to refresh the post edit URL.
  */
-function useRefreshPostEditURL(postId) {
+function useRefreshPostEditURL(postId, postEditURL) {
   const refreshPostEditURL = useCallback(() => {
-    const getPostEditURL = addQueryArgs('post.php', {
-      post: postId,
-      action: 'edit',
-    });
-    window.history.replaceState(
-      { id: postId },
-      'Post ' + postId,
-      getPostEditURL + window.location.hash
-    );
-  }, [postId]);
+    const newUrl = new URL(postEditURL);
+    newUrl.hash = window.location.hash;
+    // Remove the schema and host and just return path, params and hash.
```

Comment:
Could you elaborate on the need for this?

Reply:
The field in the api is a api is url this part of the code expects a path / param / hash. It made this fail https://github.com/google/web-stories-wp/runs/3045283496#step:7:166

Using a url class object is used to string, the https://www.example.com out of the url and add a hash in a safe way.

Reply:
Hmm sounds like this could be wrapped in a `try {} catch {}` to catch the `SecurityError` **_or_** we compare the origings of `window.location` and `postEditURL` before trying to call `history.replaceState`

Reply:
There are other places that don't wrap new url in try / catch 

https://github.com/google/web-stories-wp/blob/9edf5fc1f38933a0bd33ec885ba003524d6abea0/assets/src/edit-story/components/header/buttons/preview.js#L65-L70

https://github.com/google/web-stories-wp/blob/da23d65cbca76c604350464f0538115d280a7a06/assets/src/edit-story/app/media/media3p/api/apiFetcher.js#L187-L191

I can add one here. We can't use `window.location` as in multisite, you can have subdirectory and that starts to get messy.

Reply:
Added try / catch https://github.com/google/web-stories-wp/pull/8268/commits/6359da345b7ec8d13ecbb02db58a9b5b8ca02c52

Reply:
> There are other places that don't wrap new url in try / catch

The `history.replaceState` would need to be wrapped, as it could trigger the `SecurityError`

Reply:
Fixed.